superset/assets/javascripts/components/ColumnTypeLabel.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  type: PropTypes.string.isRequired,
+};
+
+export default function ColumnTypeLabel({ type }) {
+  let stringIcon;
+  let iconSize = '13';
+  if (type === '' || type === 'expression') {
+    stringIcon = 'ƒ';
+  } else if (type.match(/.*char.*/i) || type.match(/string.*/i) || type.match(/.*text.*/i)) {
+    stringIcon = 'ABC';
+    iconSize = '11';
+  } else if (type.match(/.*int.*/i) || type === 'LONG' || type === 'DOUBLE') {
+    stringIcon = '#';
+  } else if (type.match(/.*bool.*/i)) {
+    stringIcon = 'T/F';
+  } else if (type.match(/.*time.*/i)) {
+    stringIcon = 'time';
+  } else {
+    stringIcon = '?';
+  }
+
+  const typeIcon = stringIcon === 'time' ? <i className="fa fa-clock-o text-muted type-label" /> : (
+    <div className="text-muted type-label" style={{ fontSize: iconSize }}>{stringIcon}</div>);
```

Comment:
is the 2px `ABC` font size off enough that it warrants inline styles? ideally avoided since you're already using a new css class. I'd say a slightly larger font consistently across all icons (like 14px?) would be ideal aesthetically. I think they'd be differentiated enough from the dropdown text with the muted style you apply.

Reply:
This is what it looks like if they are the same size. To me it seemed to compete a bit with the text next to it, but I don't have strong opinions.
![image](https://user-images.githubusercontent.com/817955/37122428-5bcd1b68-2215-11e8-9c7c-0a217d017bc6.png)

Reply:
To me, keeping the same default font for ABC seemed to compete a bit with the text next to it, but I don't have strong opinions.
![image](https://user-images.githubusercontent.com/817955/37122650-3dfd8dc4-2216-11e8-8742-e2c5b6c80db4.png)